src/main/java/rs/teslaris/core/harvester/OAIPMHHarvester.java

```diff
@@ -0,0 +1,158 @@
+package rs.teslaris.core.harvester;
+
+import java.io.StringReader;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+import javax.net.ssl.SSLContext;
+import javax.xml.XMLConstants;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParserFactory;
+import javax.xml.transform.sax.SAXSource;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.TrustStrategy;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import rs.teslaris.core.harvester.common.OAIPMHDataSet;
+import rs.teslaris.core.harvester.common.OAIPMHResponse;
+import rs.teslaris.core.harvester.common.ResumptionToken;
+import rs.teslaris.core.util.exceptionhandling.exception.CantConstructRestTemplateException;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class OAIPMHHarvester {
+
+    private final String BASE_URL = "https://cris.uns.ac.rs/OAIHandlerTeslaRIS";
```

Comment:
Can we discuss how to reuse this harvester for other OAI-PMH endpoints? This line should be replaced with something configurable or we should rename OAIPMHHarvester class to reflect it is used only for cris endpoint